sphinxmixcrypto/client.py

```diff
@@ -79,21 +97,39 @@ def create_header(params, route, pki, dest, message_id, rand_reader):
         beta = xor(message_id + gamma + beta[:(2 * params.max_hops - 1) * SECURITY_PARAMETER],
                    stream_cipher.generate_stream(stream_key, params.beta_cipher_size)[:(2 * params.max_hops + 1) * SECURITY_PARAMETER])
         gamma = digest.hmac(digest.create_hmac_key(asbtuples[i]['s']), beta)
-    return (asbtuples[0]['alpha'], beta, gamma), [y['s'] for y in asbtuples]
+    sphinx_header = SphinxHeader(asbtuples[0]['alpha'], beta, gamma)
+    return sphinx_header, [y['s'] for y in asbtuples]
+
+
+def create_forward_message(params, route, pki, dest, plaintext_message, rand_reader):
+    """
+    Create a new SphinxPacket, a forward message.
```

Comment:
the create_header is used elsewhere to create a "reply block" which is not the same as a sphinx packet. so create_forward_message could be a classmethod for SphinxPacket while create_header could also be a classmethod to the SphinxHeader class. :)